### ISSUE TITLE
Feature Request - Player-defined extra template worlds

### ISSUE THREAD

**Issue by Chilly-roll** [Wilyicaro/Legacy-Minecraft#18]
Essentially, allowing people to be able to add worlds to a directory to be able to create those worlds again and again like how the mod currently lets you generate the TU69 tutorial world.
So if people wanted to, they could add the ability themselves to generate (for example) the Legacy Console mashup worlds by adding them to a "template_world" folder in their instance or something similar and have them appear in the create tab as extra options. Could be a fun extra customization option as players could also use it to have challenge maps (like skyblock) as an easily startable/restartable thing without having to tab out of the game to copy a new instance of that world into their saves.

**Comment by Wilyicaro** [comment]
It's a really good idea, I was just planning on adding all the worlds and resource packs from the mashups, and it's even better because it makes things easier on copyright since it will just be optional and downloaded separately

**Comment by Wilyicaro** [comment]
Idk why I closed

**Comment by Chilly-roll** [comment]
Including any 4J dlc content does sorta shift the mod into a gray area, since it's still available through bedrock's marketplace; hence why I suggested it's something for players to do themselves so neither you or the mod get in trouble

**Comment by SiDjackWasTaken** [comment]
I don’t think the copyright really matters to be honest unless this mod becomes really popular

**Comment by SiDjackWasTaken** [comment]
Also as an addition maybe we should add the other tutorial worlds

**Comment by AgentMindStorm** [comment]
Please avoid including any DLCs in the mod. That is the kind of thing which will get your mod cease & desisted immediately.

**Comment by Quartzmaven** [comment]
Good news, it's added in the new update!

**Comment by Chilly-roll** [comment]
Well that's good! Is there a specific folder I'm supposed to put .zips renamed to .mcsave into? The saves folder doesn't cut it and the game doesn't generate any template folder.. am I missing something here?

**Comment by Wilyicaro** [comment]
You can define what folder you want, use as example the actual world_templates.json present in the mod source https://github.com/Wilyicaro/Legacy-Minecraft/blob/master/common/src/main/resources/assets/legacy/world_templates.json

**Comment by Dragon-Mom** [comment]
Is there more info on this anywhere? I don't see anything on this feature or how to use it on either of the mod pages

**Comment by Wilyicaro** [comment]
> Is there more info on this anywhere? I don't see anything on this feature or how to use it on either of the mod pages and couldn't figure anything out based on the linked .json either

Simply add more entries to the JSON following the tutorial entry as an example, or you can create the "world_templates.json" file in other namespaces (directories in assets, such as "minecraft"), but the order is not guaranteed, it may behind the tutorial button, this will still be fixed.
If you still don't understand what each key of the json entry means, I'll use the one from the tutorial as an example:
   "legacy.menu.play_tutorial" --> ([translatable](https://minecraft.fandom.com/wiki/Resource_pack#Language) component key used in button name): {
     "icon" --> (location of the button icon sprite, note that sprites are basically textures in the [namespace:textures/gui/sprites](https://minecraft.wiki/w/Resource_pack) directory): "legacy:creation_list/tutorial",
     "templateLocation" --> (world file in .zip or .mcsave, which are the same format): "legacy:tutorial/tutorial.mcsave",
     "folderName" --> (name of the template world directory when created): "Tutorial",
     "directJoin" --> (if true, ignores the Load Save screen and enters the world directly, as in the tutorial): true
   }

**Comment by Dragon-Mom** [comment]
That makes a lot more sense, ty. I would definitely put this on the mod pages if possible, this feature is major enough to be a mod all on it's own tbh!

**Comment by Wilyicaro** [comment]
> That makes a lot more sense, ty. I would definitely put this on the mod pages if possible, this feature is major enough to be a mod all on it's own tbh!
> 
> Is it also possible to have a resource pack associated with them by default similar to the Mash-Up packs?

Of course, just save the world with the selected resource pack you want
